packages/db/index.js

```diff
@@ -506,24 +506,40 @@ export class DBClient {
     }
   }
 
+  /**
+   * @typedef {Object} listUploadReturn
```

Comment:
Can we define this type in `db-client-types.ts`, and import it where needed?

We need to make sure not only the functions annotations are updated here, but also in `index.d.ts`.

I'd keep the same convention where the type is a plain (not promise one).
and the function returns `Promise<thetype>`

nit. should be capitalised, and maybe called `ListUploadOutput` for consistency.